src/mlpack/methods/kernel_svm/kernel_svm_impl.hpp

```diff
@@ -0,0 +1,299 @@
+/**
+ * @file methods/kernel_svm/kernel_svm_impl.hpp
+ * @author Himanshu Pathak
+ *
+ * Implementation of Kernel SVM.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_KERNEL_SVM_KERNEL_SVM_IMPL_HPP
+#define MLPACK_METHODS_KERNEL_SVM_KERNEL_SVM_IMPL_HPP
+
+// In case it hasn't been included yet.
+#include "kernel_svm.hpp"
+
+namespace mlpack {
+namespace svm {
+
+template <typename MatType, typename KernelType>
+KernelSVM<MatType, KernelType>::KernelSVM(
+    const MatType& data,
+    const arma::Row<size_t>& labels,
+    const double regularization,
+    const bool fitIntercept,
+    const size_t max_iter,
+    const double tol,
+    const KernelType& kernel) :
+    regularization(regularization),
+    fitIntercept(fitIntercept),
+    kernel(kernel)
+{
+  intercept = 0;
+  Train(data, labels, max_iter, tol);
+}
+
+template <typename MatType, typename KernelType>
+KernelSVM<MatType, KernelType>::KernelSVM(
+    const size_t inputSize,
+    const double regularization,
+    const bool fitIntercept,
+    const KernelType& kernel) :
+    inputSize(inputSize),
+    regularization(regularization),
+    fitIntercept(fitIntercept),
+    kernel(kernel)
+{
+  intercept = 0;
+}
+
+template<typename MatType, typename KernelType>
+double KernelSVM<MatType, KernelType>::Train(
+    const MatType& data,
+    const arma::Row<size_t>& labels,
+    const size_t max_iter,
+    const double tol)
+{
+  arma::Row<int> label(data.n_cols);
```

Comment:
Could you rename this to `trainLabels`? Seems difficult to distinguish between `labels` and `label`

Reply:
Done